src/coreclr/vm/jitinterface.cpp

```diff
@@ -1570,10 +1580,13 @@ void CEEInfo::getFieldInfo (CORINFO_RESOLVED_TOKEN * pResolvedToken,
                 fieldAccessor = CORINFO_FIELD_STATIC_SHARED_STATIC_HELPER;
 
                 pResult->helper = getSharedStaticsHelper(pField, pFieldMT);
-
-#ifdef HOST_WINDOWS
-#ifndef TARGET_ARM
-                // For windows, we convert the TLS access to the optimized helper where we will store
+#if defined(TARGET_ARM)
+                // Optimization is disabled for linux/windows arm
+#elif !defined(_MSC_VER) && defined(TARGET_32BIT)
```

Comment:
```suggestion
#elif !defined(TARGET_WINDOWS) && defined(TARGET_X86)
```
We use `_MSC_VER` for situations where the code is specific to Microsoft C++ compiler. It is not the case here.

Reply:
(Multiple places)